apps/OboeTester/app/src/main/java/com/mobileer/oboetester/TestDataPathsActivity.java

```diff
@@ -403,11 +405,13 @@ protected String whyShouldTestBeSkipped() {
                 && (requestedInConfig.getDeviceId() != actualInConfig.getDeviceId())) {
             why += "inDev(" + requestedInConfig.getDeviceId()
                     + "!=" + actualInConfig.getDeviceId() + "),";
+            skipRemainingTests = true; // the device must have been unplugged
         }
         if (requestedOutConfig.getDeviceId() != 0
                 && (requestedOutConfig.getDeviceId() != actualOutConfig.getDeviceId())) {
             why += ", outDev(" + requestedOutConfig.getDeviceId()
                     + "!=" + actualOutConfig.getDeviceId() + "),";
+            skipRemainingTests = true; // the device must have been unplugged
         }
         if ((requestedInConfig.getInputPreset() != actualInConfig.getInputPreset())) {
             why += ", inPre(" + requestedInConfig.getInputPreset()
```

Comment:
When would the requested input preset be different than the actual input preset? Should we add a similar check here in another PR?

Reply:
The QuirksManager may substitute the InputPreset if it is broken on a device.

Reply:
Also VoicePerformance will get coerced to VoiceRecognition on older devices.
The key idea here is that if we are trying to test one Preset and got a different one then we should skip the test.

> Should we add a similar check here in another PR?

What should we add a check for?

Reply:
Thanks for the explanation. I don't think we need another check.